dotcom-rendering/src/web/components/Lightbox.importable.tsx

```diff
@@ -0,0 +1,125 @@
+import { css } from '@emotion/react';
+import { space } from '@guardian/source-foundations';
+import { useState } from 'react';
+import { getIslandsByName } from '../browser/islands/getIslandsByName';
+import { getProps } from '../browser/islands/getProps';
+import { getZIndex } from '../lib/getZIndex';
+import { LightboxPicture } from './LightboxPicture';
+import type { Props as LightboxPictureProps } from './LightboxPicture';
+
+const isServer = typeof window === 'undefined';
+
+type Props = LightboxPictureProps & {
+	caption?: string;
+};
+
+const getLightboxElements = (): Props[] => {
+	const elements = getIslandsByName('Lightbox');
+	return elements.map((element) => getProps(element) as Props);
+};
+
+export const Lightbox = (initialElementProps: Props) => {
+	// TODO: This should only be fetched once on component load
+	const elements = isServer ? [] : getLightboxElements();
+	const [index, setIndex] = useState(
+		elements.findIndex(
+			(element) => element.master === initialElementProps.master,
+		),
+	);
+	const [isOpen, setOpen] = useState(true);
+
+	const canPrev = () => index > 0;
+	const canNext = () => index < elements.length - 1;
+
+	const { master, alt, caption } =
+		index !== -1 ? elements[index] : initialElementProps;
+
+	return (
+		<>
+			<button type="button" onClick={() => setOpen(true)}>
+				Open Lightbox
+			</button>
```

Comment:
I'm interested if there's a way to make `deferUnti=interaction` more intuitive, but I haven't had the time to look into it yet